### PR TITLE
chore: pester tests

### LINKED ISSUE #167: Add a `ReadMe.md` file for the pester testcases added
### Code of Conduct

- [X] I have read and agree to the project's Code of Conduct.
- [X] Vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue initial description to help the maintainers prioritize.
- [X] Do not leave "+1" or other comments that do not add relevant information or questions.
- [X] If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### Description

There is a need to understand what is pester, how to use pester and so on. This is needed to use it  for testing the power shell code. This requires a readme doc to be made available for the user. 

### Use Case(s)

Provide information on 
1. How to run the suite entirely
2. How to run individual testcases
3. How to run only positive cases, or only negative cases
4. Where logs are stored

### Potential Configuration

Provide information on 

1. How to run the suite entirely
    `Invoke-Pester <suite name>`
2. How to run individual testcases
    `Invoke-Pester -Tag <tag assosiated with the testcase>`
3. How to run only positive cases, or only negative cases
    `Invoke-Pester -Tag  <"Positive"/"Negative">`
4. Where logs are stored
     ./tests/logs folder
5. Where sample pester files are stored
    ./tests

### References

_No response_

### PR DESCRIPTION
**Pull request by tenthirtyam**
### Summary

- Adds documentation for use of the Pester tests.
- Renames `InputJSON.json` to `inputJson.json.example`.
- Renames objects for the JSON to be more accruate.
- Updates `CONTRIBUTING.md` regarding testing additions.
- Minor typo corrections.

### Type

- [ ] Bugfix
- [ ] Enhancement or Feature
- [ ] Code Style or Formatting
- [ ] Documentation
- [ ] Refactoring
- [x] Chore
- [ ] Other
        Please describe:

### Breaking Changes?

- [ ] Yes, there are breaking changes.
- [x] No, there are no breaking changes.

### Test and Documentation

- [ ] Tests have been completed.
- [x] Documentation has been added or updated.

Asking @sowjuec to review and confirm testing based on the JSON adjustments.

### Issue References

Closes #167

### CODE REVIEW

File: tests/PPM.Tests.ps1
Comment (sowjuec) [inline tests/PPM.Tests.ps1:1]: In the file `${$inputData.$domain.'Manager Node'[0]}` can be replaced with `$nsxManagerNode` variable and also `${$inputData.$domain.'Edge Nodes'[0]}` can be replaced with `$nsxEdgeNode` variable

File: tests/inputData.json.example
Comment (sowjuec) [inline tests/inputData.json.example:1]: Json file, the domain name is wrong like should have been "sfo-w01" instead of "sfo-mw01" in the Domains key.

**Review by sowjuec — CHANGES_REQUESTED** [review]
In the example Json file, the domain name is wrong like should have been "sfo-w01" instead of "sfo-mw01" in the Domains key.

In the PPM.Tests.ps1 file ${$inputData.$domain.'Manager Node'[0]} can be replaced with $nsxManagerNode variable and also ${$inputData.$domain.'Edge Nodes'[0]} can be replaced with $nsxEdgeNode variable

File: tests/inputData.json.example
Comment (tenthirtyam) [reply tests/inputData.json.example:1]: Forced pushed a change in https://github.com/vmware/powershell-module-for-vmware-cloud-foundation-password-management/commit/4d4d706efe6d943d0e048dc45b56723c64f6d736.

File: tests/PPM.Tests.ps1
Comment (tenthirtyam) [reply tests/PPM.Tests.ps1:1]: Forced pushed a change in https://github.com/vmware/powershell-module-for-vmware-cloud-foundation-password-management/commit/4d4d706efe6d943d0e048dc45b56723c64f6d736.

**Comment by tenthirtyam** [comment]
Hi @sowjuec 👋🏻 - I forced pushed a change in https://github.com/vmware/powershell-module-for-vmware-cloud-foundation-password-management/commit/4d4d706efe6d943d0e048dc45b56723c64f6d736 for your review comments. Take a look and let me know how it looks to you.

Ryan

**Comment by sowjuec** [comment]
> Hi @sowjuec 👋🏻 - I forced pushed a change in [4d4d706](https://github.com/vmware/powershell-module-for-vmware-cloud-foundation-password-management/commit/4d4d706efe6d943d0e048dc45b56723c64f6d736) for your review comments. Take a look and let me know how it looks to you.
> 
> Ryan

Looks good Ryan

**Review by sowjuec — APPROVED** [review]

**Pull request merged**